src/components/GlobalPreferences/Notifications/NotificationsLogin.js

```diff
@@ -0,0 +1,173 @@
+import React, { useState, useCallback } from 'react'
+import PropTypes from 'prop-types'
+import {
+  Box,
+  Button,
+  GU,
+  TextInput,
+  Info,
+  IconMail,
+  IconCross,
+  IconCheck,
+  textStyle,
+  useTheme,
+} from '@aragon/ui'
+import { login } from './notification-service-api'
+import { network } from '../../../environment'
+import { validateEmail } from '../../../utils'
+import notificationSvg from './notifications.svg'
+import throttle from 'lodash.throttle'
+
+export default function NotificationsLogin({ dao, authState, onEmailChange }) {
+  const [inputEmail, setInputEmail] = useState('')
+  const [emailInvalid, setEmailInvalid] = useState(null)
+  const [emailBlured, setEmailBlured] = useState(false)
+  const [apiError, setApiError] = useState(null)
+  // The notifications API expects mainnet or rinkeby. This deviates from web3's getNetworkType which returns main
+  const ethNetwork = network.type === 'main' ? 'mainnet' : 'rinkeby'
+
+  const handleEmailValidation = useCallback(
+    throttle(email => {
+      if (email.length > 3 && email.includes('@')) {
+        setEmailInvalid(!validateEmail(email))
+      }
+    }, 600),
+    [setEmailInvalid]
+  )
+  const handleEmailBlur = useCallback(
+    e => {
+      setEmailBlured(true)
+      const email = e.target.value
+      setEmailInvalid(!validateEmail(email))
+    },
+    [setEmailBlured, setEmailInvalid]
+  )
+
+  const handleEmailChange = useCallback(
+    e => {
+      const email = e.target.value
+      setInputEmail(email)
+      handleEmailValidation(email)
+    },
+    [handleEmailValidation, setInputEmail]
+  )
+
+  const handleLogin = async e => {
+    e && e.preventDefault()
+    try {
+      await login({ email: inputEmail, dao, network: ethNetwork })
+      onEmailChange(inputEmail)
+    } catch (e) {
+      setApiError(e.message)
+      console.error('Failed to login', e)
+    }
+  }
+
+  const theme = useTheme()
+
+  return (
+    <Box heading="Email notifications">
+      <NotificationImage />
+      <form onSubmit={handleLogin}>
+        <div
+          css={`
+            display: grid;
+            align-items: end;
+            grid-template-columns: 1fr auto;
+            grid-gap: ${GU}px ${2 * GU}px;
+            margin-bottom: ${2 * GU}px;
+          `}
+        >
+          <label
+            css={`
+              color: ${theme.surfaceContentSecondary};
+              display: block;
+            `}
+          >
+            Email address
+            <TextInput
+              css={`
+                border-color: ${emailInvalid === true
+                  ? theme.error
+                  : theme.border};
+              `}
+              adornment={
+                emailInvalid === false ? (
+                  <IconCheck color={theme.success} />
+                ) : (
+                  <IconCross color={theme.error} />
```

Comment:
First render shows the error state.

```suggestion
: inputEmail.trim() ? (
  <IconCross color={theme.error} />                                                                                                            
) : null
```

Reply:
And, odd thing when I start typing (only on first char) the input looses focus

Reply:
![Input email](https://user-images.githubusercontent.com/3072458/63685486-c2c83780-c7ff-11e9-8960-84fd30cb2653.gif)

Couple of things shown here:
- [ ] The comment from above about losing focus
- [ ] Right away when I click on `Sign in` I get the error label
- [ ] Cannot click on the button (it is disabled) until I lose focus on the input box

Reply:
> And, odd thing, when I start typing (only on first char) the input looses focus

I noticed this too. This happens when we re-render the `TextInput` component with different `adornment` prop values. 

This feels like something that should be handled by `TextInput`. Any thoughts?

Reply:
@bpierre I do think `input` should take care of not losing focus here, it is not an odd feature request: the use case of showing a different adornment after validation. Thoughts?

Reply:
Ah yes we should fix that. I opened an issue https://github.com/aragon/aragon-ui/issues/534

Reply:
>  Right away when I click on Sign in I get the error label

Interesting. What error do you see in the network tab?

Reply:
The `color` prop is not supported on icons, instead it inherits from `color` in CSS.

```jsx
<IconCross css={`color: ${theme.error}`} />
```

But it is something we should definitely add now that we have monochromatic icons :+1: 

Something else is that `theme.negative` should be used instead of `theme.error` ([`error` / `success`](https://github.com/aragon/aragon-ui/blob/newstyle/src/theme/theme-light.js#L92-L94) have been renamed `positive` / `negative` to be more generic).

Reply:
> Interesting. What error do you see in the network tab?
CORS issue, prob because of `localhost`:

![image](https://user-images.githubusercontent.com/3072458/63691337-3160c180-c80f-11e9-9acc-21807fa177aa.png)

Reply:
We cannot enable CORS for localhost. It's best to use the now deployment. 

Comments addressed besides the fix [awaiting](https://github.com/aragon/aragon-ui/issues/534)

Reply:
Thanks for clarifying :)